DataFormats/Detectors/TPC/include/DataFormatsTPC/CalibdEdxTrackTopologyPol.h

```diff
@@ -49,71 +73,114 @@ class CalibdEdxTrackTopologyPol
   /// \param z z position of the cluster
   /// \param relPad absolute relative pad position of the track
   /// \param relTime relative time position of the track
-  GPUd() float getCorrection(const int region, const ChargeType charge, const float tanTheta, const float sinPhi, const float z, const float relPad, const float relTime) const
+  GPUd() float getCorrection(const int region, const ChargeType charge, const float tanTheta, const float sinPhi, const float z, const float relPad, const float relTime, const float threshold = 0) const
   {
-    const auto& param = mParams[regionIndex(region, charge)];
-    const float x[FXDim]{tanTheta, sinPhi, z, relPad, relTime};
-    const float corr = evalPol4_5D(x, param);
+    const float x[]{tanTheta, sinPhi, z, relPad, relTime, threshold};
+    const float corr = charge == ChargeType::Tot ? mCalibPolsqTot[region].eval(x) : mCalibPolsqMax[region].eval(x);
```

Comment:
Perhaps
```suggestion
    const float corr = (charge == ChargeType::Tot) ? mCalibPolsqTot[region].eval(x) : mCalibPolsqMax[region].eval(x);
```
for readability.